File: Student_Room_Swapping/backend/src/main/java/org/tushardubey/java/backend/dto/SwapRequest.java
package org.tushardubey.java.backend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class SwapRequest {
    @NotNull
    private Long applicantId;

    @NotNull
    private Long recipientId;

    @NotBlank
    private String applicationMessage;
}

File: Student_Room_Swapping/frontend/src/models/SwapRequest.js
// src/model/SwapRequest.js
export default class SwapRequest {
    constructor(data) {
        this.applicantId = data.applicantId || ""; // Logged-in user ID
        this.recipientId = data.recipientId || ""; // Recipient ID
        this.applicationMessage = data.applicationMessage || ""; // Message to recipient
       
    }
}
